iceoryx2/tests/service_tests.rs

```diff
@@ -53,6 +63,230 @@ mod service {
+        fn open(service_name: &ServiceName) -> Result<Self::Factory, Self::OpenError>;
+    }
+
+    impl<Sut: Service> SutFactory for publish_subscribe::PortFactory<Sut, u64> {
+        type Factory = publish_subscribe::PortFactory<Sut, u64>;
+        type CreateError = PublishSubscribeCreateError;
+        type OpenError = PublishSubscribeOpenError;
+
+        fn create(service_name: &ServiceName) -> Result<Self::Factory, Self::CreateError> {
+            Sut::new(&service_name).publish_subscribe().create::<u64>()
+        }
+
+        fn open(service_name: &ServiceName) -> Result<Self::Factory, Self::OpenError> {
+            Sut::new(&service_name).publish_subscribe().open::<u64>()
+        }
+    }
+
+    impl<Sut: Service> SutFactory for event::PortFactory<Sut> {
+        type Factory = event::PortFactory<Sut>;
+        type CreateError = EventCreateError;
+        type OpenError = EventOpenError;
+
+        fn create(service_name: &ServiceName) -> Result<Self::Factory, Self::CreateError> {
+            Sut::new(&service_name).event().create()
+        }
+
+        fn open(service_name: &ServiceName) -> Result<Self::Factory, Self::OpenError> {
+            Sut::new(&service_name).event().open()
+        }
+    }
+
+    fn concurent_creating_services_with_unique_names_is_successful_impl<Factory: SutFactory>() {
+        let _watch_dog = Watchdog::new_with_timeout(Duration::from_secs(30));
+        const NUMBER_OF_THREADS: usize = 4;
+        const NUMBER_OF_ITERATIONS: usize = 50;
+
+        let barrier_enter = Barrier::new(NUMBER_OF_THREADS);
+        let barrier_exit = Barrier::new(NUMBER_OF_THREADS);
+
+        std::thread::scope(|s| {
+            let mut threads = vec![];
+            for _ in 0..NUMBER_OF_THREADS {
+                threads.push(s.spawn(|| {
+                    for _ in 0..NUMBER_OF_ITERATIONS {
+                        barrier_enter.wait();
+
+                        let service_name = generate_name();
+                        let _sut = Factory::create(&service_name).unwrap();
+
+                        barrier_exit.wait();
+                    }
+                }));
+            }
+
+            for thread in threads {
+                thread.join().unwrap();
+            }
+        });
+    }
+
+    #[test]
+    fn concurrent_creating_services_with_unique_names_is_successful<Sut: Service>() {
+        concurent_creating_services_with_unique_names_is_successful_impl::<
+            publish_subscribe::PortFactory<Sut, u64>,
+        >();
+
+        concurent_creating_services_with_unique_names_is_successful_impl::<event::PortFactory<Sut>>(
+        );
+    }
+
+    fn concurrent_creating_services_with_same_name_fails_for_all_but_one_impl<
+        Factory: SutFactory,
+        F: Sync,
+    >(
+        error_check: F,
+    ) where
+        F: Fn(Factory::CreateError),
+    {
+        let _watch_dog = Watchdog::new_with_timeout(Duration::from_secs(30));
+        const NUMBER_OF_THREADS: usize = 4;
```

Comment:
What do you think about maximizing the used threads to the number of cores available, which should be at least 2? This is what I do for other concurrent test cases. Yes, I know it then depends on the machine it is running, but my thinking was that the more meaningfully stressed a test is, the more likely a race condition reveals itself —and this is what we ultimately want.

So you would rewrite this to:
```rust
let number_of_threads = (SystemInfo::NumberOfCpuCores.value()).clamp(2, usize::MAX);
```

Reply:
I just noticed locally that the really intersting stuff happens when the number of threads is around two times the number of logical cores